### PR TITLE
new file:   packages/omlr/omlr.1.0.3/opam

### PR DESCRIPTION
**Pull request by UnixJunkie**
the dune-project file was missing in order to compile w/ a recent version of dune; should compile w/ ocaml-5 and others

### CODE REVIEW

File: packages/omlr/omlr.1.0.3/opam [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+homepage: "https://github.com/UnixJunkie/omlr"
+bug-reports: "https://github.com/UnixJunkie/omlr/issues"
+dev-repo: "git+https://github.com/UnixJunkie/omlr.git"
+license: "BSD-3-Clause"
+build: ["dune" "build" "-p" name "-j" jobs]
+depends: [
+  "batteries"
+  "conf-gnuplot"
+  "conf-r"
+  "cpm"
```
Comment (kit-ty-kate) [inline packages/omlr/omlr.1.0.3/opam:13]:
```suggestion
  "cpm" {>= "4.0.0"}
```

**Comment by UnixJunkie** [comment]
version constraint on cpm reflected upstream

File: packages/omlr/omlr.1.0.3/opam [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+homepage: "https://github.com/UnixJunkie/omlr"
+bug-reports: "https://github.com/UnixJunkie/omlr/issues"
+dev-repo: "git+https://github.com/UnixJunkie/omlr.git"
+license: "BSD-3-Clause"
+build: ["dune" "build" "-p" name "-j" jobs]
+depends: [
+  "batteries"
+  "conf-gnuplot"
+  "conf-r"
+  "cpm" {>= "4.0.0"}
```
Comment (kit-ty-kate) [inline packages/omlr/omlr.1.0.3/opam:13]:
```suggestion
  "cpm" {>= "7.0.0"}
```
Uses `Cpm.Utls`

**Comment by UnixJunkie** [comment]
updated new cpm version constraint upstream

File: packages/omlr/omlr.1.0.3/opam [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+homepage: "https://github.com/UnixJunkie/omlr"
+bug-reports: "https://github.com/UnixJunkie/omlr/issues"
+dev-repo: "git+https://github.com/UnixJunkie/omlr.git"
+license: "BSD-3-Clause"
+build: ["dune" "build" "-p" name "-j" jobs]
+depends: [
+  "batteries"
+  "conf-gnuplot"
+  "conf-r"
+  "cpm" {>= "7.0.0"}
```
Comment (kit-ty-kate) [inline packages/omlr/omlr.1.0.3/opam:13]:
```suggestion
  "cpm" {>= "10.2.1"}
```
```
#=== ERROR while compiling omlr.1.0.3 =========================================#
# context              2.2.0~alpha~dev | linux/x86_64 | ocaml-base-compiler.4.14.1 | pinned(https://github.com/UnixJunkie/omlr/archive/v1.0.3.tar.gz)
# path                 ~/.opam/4.14/.opam-switch/build/omlr.1.0.3
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune build -p omlr -j 255
# exit-code            1
# env-file             ~/.opam/log/omlr-7-9da287.env
# output-file          ~/.opam/log/omlr-7-9da287.out
### output ###
#       ocamlc src/.model.eobjs/byte/model.{cmi,cmo,cmt} (exit 2)
# (cd _build/default && /home/opam/.opam/4.14/bin/ocamlc.opt -w -40 -g -bin-annot -I src/.model.eobjs/byte -I /home/opam/.opam/4.14/lib/batteries -I /home/opam/.opam/4.14/lib/bytes -I /home/opam/.opam/4.14/lib/cpm -I /home/opam/.opam/4.14/lib/dolog -I /home/opam/.opam/4.14/lib/minicli -I /home/opam/.opam/4.14/lib/num -I /home/opam/.opam/4.14/lib/ocaml/threads -I src/.omlr.objs/byte -no-alias-deps -o src/.model.eobjs/byte/model.cmo -c -impl src/model.ml)
# File "src/model.ml", line 62, characters 4-29:
# 62 |     Cpm.Utls.train_test_split train_portion all_lines in
#          ^^^^^^^^^^^^^^^^^^^^^^^^^
# Error: The module Cpm.Utls is an alias for module Cpm__Utls, which is missing
```

**Comment by samoht** [comment]
More lower-bounds issues (with cpm?) -- @UnixJunkie do you mind looking at those?

image

File: packages/omlr/omlr.1.0.3/opam [hunk 4]
```diff
@@ -0,0 +1,40 @@
+opam-version: "2.0"
+authors: "Francois Berenger"
+maintainer: "unixjunkie@sdf.org"
+homepage: "https://github.com/UnixJunkie/omlr"
+bug-reports: "https://github.com/UnixJunkie/omlr/issues"
+dev-repo: "git+https://github.com/UnixJunkie/omlr.git"
+license: "BSD-3-Clause"
+build: ["dune" "build" "-p" name "-j" jobs]
+depends: [
+  "batteries"
```
Comment (mseri) [inline packages/omlr/omlr.1.0.3/opam:10]:
```suggestion
  "batteries" {>= "3.0.0"}
```
This is where `count_char` was introduced

**Comment by mseri** [comment]
This now builds fine on all targets. The failures are due to missing system packages for R

**Pull request merged**